transfer/adapterbase.go

```diff
@@ -124,11 +124,18 @@ func (a *adapterBase) worker(workerNum int, ctx interface{}) {
 		}
 		tracerx.Printf("xfer: adapter %q worker %d processing job for %q", a.Name(), workerNum, t.Object.Oid)
 
+		// tt is the time that we are to compare the transfer's
+		// `expired_at` property against.
+		tt := time.Now().Add(-objectExpirationGracePeriod)
+
 		// Actual transfer happens here
 		var err error
-		if t.Object.IsExpired(time.Now().Add(objectExpirationGracePeriod)) {
+		if expAt, expired := t.Object.IsExpired(tt); expired {
 			tracerx.Printf("xfer: adapter %q worker %d found job for %q expired, retrying...", a.Name(), workerNum, t.Object.Oid)
-			err = errors.NewRetriableError(errors.Errorf("lfs/transfer: object %q has expired", t.Object.Oid))
+			err = errors.NewRetriableError(errors.Errorf(
+				"lfs/transfer: object %q has expired at %s, %s ago",
```

Comment:
Maybe just 'expires at' rather than trying to give a time offset since in anything expiring in the very near future is not worth trying and the 'ago' could look backwards - it's reasonable to reject a link that expires in 1s so we fetch new ones from the API but `-1s ago` would look weird ;)

Worth noting these messages should never be seen by the user anyway. The expected behaviour is that they trigger another API request for fresh links internally.